supabase-openai-summary/src/steps/summarizers.py

```diff
@@ -0,0 +1,75 @@
+#  Copyright (c) ZenML GmbH 2023. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+
+from typing import List
+import openai
+
+from zenml.steps import step, BaseParameters, StepContext
+from zenml.client import Client
+from zenml.post_execution import get_pipeline, PipelineRunView
+
+
+class SummarizerParams(BaseParameters):
+    """Prompts for the summarizer."""
+
+    system_content: str = "Act like a data analytics expert."
+    prompt_preamble: str = "Summarize the latest data from the database. The data is YouTube video titles and we are keen to understand what sort of videos users are accessing, the trends, and similarities between them. We also have the analysis from last time that we will give to you as input."
+    prompt_example: str = "The latest data indicates the following key insights: 1. The number of users has increased by 10% in the last 24 hours. 2. The number of users has increased by 10% in the last 24 hours. 3... Compared to our last summary, it seems our users are starting to watch more videos about cats."
+
+
+@step(enable_cache=True)
+def gpt_4_summarizer(
+    params: SummarizerParams, documents: List[str], context: StepContext
+) -> str:
+    """Summarizes the data using GPT-4."""
+    openai_secret = Client().get_secret("openai")
+
+    p = get_pipeline(context.pipeline_name)
+
+    last_step = PipelineRunView(
+        Client().list_runs(size=5, pipeline_id=p.id)[1]
```

Comment:
```suggestion
        Client().list_runs(size=5, pipeline_id=p.id)[0]
```

Reply:
@fa9r  Are you sure about this? I think 0 would be the current run